tasmota/tasmota_xsns_sensor/xsns_14_sht3x.ino

```diff
@@ -106,16 +106,15 @@ bool Sht3xRead(uint32_t sensor) {
   if ((Sht3xComputeCrc(&data[0], 2) != data[2]) || (Sht3xComputeCrc(&data[3], 2) != data[5])) {
     return false;
   }
-  float t;
-  float h;
-  t = ((((data[0] << 8) | data[1]) * 175) / 65535.0) - 45.0;
+  float t = ((((data[0] << 8) | data[1]) * 175) / 65535.0f) - 45;
```

Comment:
Wouldn't you also like to calculate with integers here?

Reply:
> Wouldn't you also like to calculate with integers here?
Hi Erik, yes, I even suggested it here: https://github.com/arendst/Tasmota/pull/21816#discussion_r1687925556 
We discussed it quite extensively afterwards and @sfromis had a valid point here: https://github.com/arendst/Tasmota/pull/21816#discussion_r1690224615
Do you have a better idea?

Reply:
Did I understand correctly that the objection only concerns accuracy? Then we just calculate with hundredths in integers and select the appropriate type so that it does not overflow.